sql/parser/sql.y

```diff
@@ -3198,20 +3200,26 @@ array_expr_list:
   }
 
 extract_list:
-  extract_arg FROM a_expr {}
-| /* EMPTY */ {}
+  extract_arg FROM a_expr
+  {
+    $$ = Exprs{StrVal($1), $3}
+  }
 
-// Allow delimited string SCONST in extract_arg as an SQL extension.
-// - thomas 2001-04-12
 extract_arg:
-  IDENT {}
-| YEAR {}
-| MONTH {}
-| DAY {}
-| HOUR {}
-| MINUTE {}
-| SECOND {}
-| SCONST {}
+  IDENT
+| YEAR
+| QUARTER
+| MONTH
+| WEEK
+| DAY
+| DAYOFWEEK
+| DAYOFYEAR
```

Comment:
As I mentioned elsewhere, I think we've gone the wrong direction making these unreserved keywords. Better to just let `DAYOFWEEK`, etc be captured by the `IDENT` rule.